projects/plugins/crm/api/create_transaction.php

```diff
@@ -27,19 +27,37 @@
 		$json_params = file_get_contents( 'php://input' );
 		$new_trans   = json_decode( $json_params, true );
 
-		// REQUIRED
-		$orderid = -1;
+// REQUIRED FIELDS
+$required_fields = array(
+	'orderid',
+	'email',
+);
+
+// Check for required fields. Fail if any are missing.
+foreach ( $required_fields as $field ) {
+	if ( ! array_key_exists( $field, $new_trans ) ) {
+		wp_send_json(
+			array(
+				'error'   => 400,
+				'message' => 'Missing required field: ' . $field,
+			)
+		);
+		exit();
+	}
+}
+
+$orderid = -1;
```

Comment:
In the current implementation (which I don't think we should merge) we wouldn't need "default required" line.